src/docker/restorableappengine.cc

```diff
@@ -963,15 +955,11 @@ void RestorableAppEngine::checkAvailableStorageInStores(const std::string& app_n
                                                         const uint64_t& docker_required_storage) const {
   auto checkRoomInStore = [&](const std::string& store_name, const uint64_t& required_storage,
                               const boost::filesystem::path& store_path) {
-    boost::uintmax_t capacity;
-    boost::uintmax_t available;
-
-    std::tie(capacity, available) = storage_space_func_(store_path);
-    LOG_INFO << app_name << " -> " << store_name << " store total update size: " << required_storage
-             << " bytes; available: " << available << " (out of " << capacity << ") "
-             << ", path: " << store_path.string();
-    if (required_storage > available) {
-      throw InsufficientSpaceError(store_name, store_path.string(), required_storage, available);
+    storage::Volume::UsageInfo usage_info{storage_space_func_(store_path)};
+    LOG_INFO << app_name << " -> " << store_name
+             << " store total update size: " << usage_info.withRequired(required_storage);
```

Comment:
don't you need a `if usage_info.isOk` before printing printing this?

Reply:
No. I've done intentionally. The `operator <<` for `UsageInfo ` has `if` if it's not ok, and will print an error. I find it helpful, since a user can see problems with obtaining usage statistics.